### DISCUSSION TITLE
Conductor 2.x to 3.x upgrade

### DISCUSSION THREAD

**Discussion by soumit09** [Netflix/conductor#2365] [Q&A] [answered]
Hi Team,
I could see in conductor 3.0.6 upgrade few api's are removed  like /tasks/in_progress/{workflowId}/{taskRefName}.

We have some use case like for aync workflow implementation we use taskRefName to fetch the details of specific task(TaskId)using above api  and mark it as success or fail using a callback api. Is there any alternate way we can fetch  task details by taskRefName?

**Comment by apanicker-nflx** [comment, marked as answer]
>Is there any alternate way we can fetch task details by taskRefName?

You could get the workflow details including tasks using the GET /workflow/{workflowId} API. Alternatively, if you have the taskId, you can get the task details using the GET /tasks/{taskId} API.

**Reply by soumit09** [reply]
Thanks for the update. I could see using /workflow/{workflowId}  i could get the details and can have workaround to overcome my issue. 
Just curious to understand what is the reason behind removing /tasks/in_progress/{workflowId}/{taskRefName} api from conductor 3.0.6. Is it due to the fact same info is present in /workflow/{workflowId} api?

**Reply by apanicker-nflx** [reply]
Part of the reason was to reduce the API footprint of Conductor which exposes similar information through multiple redundant endpoints. Another motivating factor was that some of the APIs were designed with the initial version of Conductor which relied on Dynomite persistence and is no longer applicable across all the datastore implementations.
